tools/build_api.py

```diff
@@ -250,6 +251,13 @@ def find_valid_toolchain(target, toolchain):
                 and "uARM" in {toolchain_name, target.default_toolchain}
             ):
                 end_warnings.append(UARM_TOOLCHAIN_WARNING)
+
+            if (
+                toolchain_name == "IAR"
+                and target.c_lib.lower() == "small"
```

Comment:
Since c_lib appears to be an optional attribute, need to use getattr() here to ensure that an exception is not thrown when it is not specified.